### ISSUE TITLE
Converter.php

### ISSUE THREAD

**Issue by spirillen** [r-a-y/mobile-hosts#32]
Hi @r-a-y 

I'm using your lovely `converter.php` script to import various sources into my search library https://github.com/external-sources/hosts-sources, however, do to what appears to be a bug in one of the sources https://github.com/mitchellkrogza/Phishing.Database/ where the [ALL-phishing-links.txt](https://raw.githubusercontent.com/mitchellkrogza/Phishing.Database/master/ALL-phishing-links.txt) contains all the domains from [ALL-phishing-domains.txt](https://raw.githubusercontent.com/mitchellkrogza/Phishing.Database/master/ALL-phishing-domains.txt). I've come to the need to be able to import a lists full of URI's, while I can only use the plain domains. Is this a option you could build into the script?

I would be very happy, if you could help whit this feature, as I rely on it when I investigate domain

------

Future note: Slowly, like very slowly, learning Rust language, and that project [domain-hub](https://github.com/external-sources/domain-hub) is to take over [hosts-sources](https://github.com/external-sources/hosts-sources)

**Comment by r-a-y** [comment]
Hey @spirillen , glad you find the script handy!

As for your request, I think my `converter.php` shouldn't handle URLs. That being said, I've written a PHP script for you to try: https://gist.github.com/r-a-y/7f108351ab5c62f317da6b6c10273589 . I added trailing `0.0.0.0` to each domain, but if you do not want that, you can remove or comment out lines 48-51. Let me know if that works for you.

**Comment by spirillen** [comment]
Thanks a lot @r-a-y I'll test it in a couple of days when I'm finished the transformation from ubuntu to Debian and have found my files ;) .  And no I do not use the Zeros as I only uses the flat(Domains only) files as search index with `git grep`

**Comment by spirillen** [comment]
Thanks a lot for the script @r-a-y it works like a charm,

How ever, I've found that your original script, which I have modified after my best skills, are unable to import a number hosts files. You can see them as they have been commented out in this commit

As you might notice in line 64-82 then it seems like being rather random what fails and what is a success

https://github.com/external-sources/hosts-sources/blob/master/scripts/converter.php#L64-L84

Do you have any idea to why these might be failing?

**Comment by r-a-y** [comment]
It could be that those lists are very large and causing memory issues. It might be best if I added an option to convert lists via command-line arguments instead of having an array that is hardcoded.

Then you could parse those lists one at a time via command-line. For example:

```
php converter.php https://raw.githubusercontent.com/ShadowWhisperer/BlockLists/master/Lists/Ads ShadowWhispererAds
```
And then we can figure out which lists are causing the parsing problems.

I'll look into this in a bit.

**Comment by spirillen** [comment]
That is very kind of you. I'm grateful for your help

**Comment by r-a-y** [comment]
@spirillen I've updated the `converter.php` script to allow for a filter list to be passed through the command-line. See commit 725ace7.

With the new update, you would use the following in the CLI:

```
php converter.php name=ShadowWhispererAds url=https://raw.githubusercontent.com/ShadowWhisperer/BlockLists/master/Lists/Ads
```

Let me know if that works for you and let me know which lists are causing problems once you are able to test.

**Comment by spirillen** [comment]
Super, thanks, I'll  give it a spin, probably Friday, as I'm no tat home to morrow,

I'll let you know the result.

**Comment by spirillen** [comment]
Hi @r-a-y 

Now I've been playing a bit with the modified converter. Unfortunately, it seems like it is not even trying to download the lsits.

```bash
time php -f scripts/converter.php name=adAway url=https://raw.githubusercontent.com/AdAway/adaway.github.io/master/hosts.txt
Converting adAway...
adAway converted to HOSTS file - see adAway.txt

real    0m0.146s
user    0m0.044s
sys     0m0.018s
```

![image](https://github.com/user-attachments/assets/0e41285d-6c55-4a91-9a3f-57dcb61c0568)

as you can see, it gives up after .018 seconds... Do you happens to have any ideas to why this might happen?

To compare, here is the bash results

```bash
time wget -qO- "https://raw.githubusercontent.com/AdAway/adaway.github.io/master/hosts.txt" | awk '/localhost/{next}; /^(#|$)/{ next }; { if ( $1 ~ /^[0-9]/ ) printf("%s\n",tolower($2)) | "sort -i | uniq -u -i " }' >"data/adaway/domain.list"

real    0m0.242s
user    0m0.060s
sys     0m0.020s
```

![image](https://github.com/user-attachments/assets/d5deafbb-7e03-47ec-a279-c48b60ddcfe5)

**Comment by r-a-y** [comment]
Hi @spirillen , your adAway list that you're attempting to convert is already a hosts file: https://raw.githubusercontent.com/AdAway/adaway.github.io/master/hosts.txt . This converter only works for ad filter lists.

I'm guessing you want to use the urls to domains converter I crafted for you at https://gist.github.com/r-a-y/7f108351ab5c62f317da6b6c10273589 . I've updated that script over on Github Gist. Leave a comment there if you have trouble with it.
